opencsp/app/sofast/SofastGUI.py

```diff
@@ -451,37 +442,40 @@ def _enable_btns(self) -> None:
         self.btn_live_view.config(state=state_im_acqu)
         self.btn_close_camera.config(state=state_im_acqu)
 
-    def _check_system_loaded(self) -> None:
+    def on_service_set(self):
+        super().on_service_set()
+        self._enable_btns()
+
+    def on_service_unset(self):
+        super().on_service_unset()
+        self._enable_btns()
+
+    def _check_system_loaded(self) -> bool:
         """Checks if the system class has been instantiated"""
-        if self.system is None:
-            messagebox.showerror('Error', 'Both ImageAcquisiton and ImageProjection must both be loaded.')
-            return
+        try:
+            system = self.service.system
+            return True
+        except Exception as ex:
+            messagebox.showerror('Error', repr(ex))
+            return False
```

Comment:
Same question as in previous comment:
https://github.com/sandialabs/OpenCSP/pull/61/files#r1544895791